src/multialign/BasicClasses/Ligand.hpp

```diff
@@ -0,0 +1,59 @@
+//
+// Created by chris on 11/4/23.
+//
+#pragma once
+
+#include "Forward.hpp"
+#include "UniquePoseIdentifier.hpp"
+#include <GraphMol/GraphMol.h>
+
+#include <unordered_set>
+
+namespace MultiAlign {
+
+    /**
+     * Represents a molecule.
+     */
+    class Ligand {
+    public:
+        Ligand(const RDKit::RWMol& mol,
+               const UniquePoseSet &poses,
+               LigandID id);
+
+        /**
+         * get idenitifers of all poses embedded in ligand.
+         * @return
+         */
+        [[nodiscard]] UniquePoseSet getPoses() const noexcept;
+
+        /**
+         *
+         * @return
+         */
+        [[nodiscard]] LigandID getID() const noexcept;
+
+        /**
+         *
+         * @return
+         */
+        unsigned getHeavyAtomSize() const noexcept;
+
+        /**
+         *
+         * @return The number of conformers embedded in the ligand.
+         */
+        unsigned getNofPoses() const noexcept;
+
+        /**
+         *
+         * @return The molecule represented by the ligand.
+         */
+        RDKit::RWMol getMolecule() const noexcept;
+
+    private:
+        LigandID m_id;
```

Comment:
Why are those variables called m_ ? I learned different naming convention

Reply:
thats AMD research group style. If its okay for you id appreciate if we can stick with it. lots of m_ members already defined and kind of habit :)

Reply:
Please just make it uniform